src/x509.rs

```diff
@@ -47,3 +47,26 @@ pub(crate) fn set_extension_once<T>(
         }
     }
 }
+
+pub(crate) fn remember_extension(
+    extension: &Extension,
+    mut handler: impl FnMut(u8) -> Result<(), Error>,
+) -> Result<(), Error> {
+    // ISO arc for standard certificate and CRL extensions.
+    // https://www.rfc-editor.org/rfc/rfc5280#appendix-A.2
+    static ID_CE: [u8; 2] = oid![2, 5, 29];
+
+    if extension.id.len() != ID_CE.len() + 1
+        || !extension.id.as_slice_less_safe().starts_with(&ID_CE)
+    {
+        return if extension.critical {
```

Comment:
Style nit: `match`, as elsewhere. Maybe this should be a `Extension::unsupported() -> Result<(), Error>` method? I think that method would have 4 potential call sites by the end of this PR.

Reply:
Good idea, thanks! I implemented the `Extension::unsupported()` idea and threaded its use through subsequent commits.